plugin/RunalyzePluginPanel_TagsSummary/class.RunalyzePluginPanel_TagsSummary.php

```diff
@@ -0,0 +1,187 @@
+			$Configuration->update('sport');
+			Cache::delete(PluginConfiguration::CACHE_KEY);
+		}
+
+		$Timescale = new PluginConfigurationValueSelect('tagssummary_timescale', __('Show tags for this period of months'));
+		$Timescale->setOptions($this->Timescale);
+
+		$Configuration->addValue($Timescale);
+
+		if (isset($_GET['tagssummary_timescale']) && in_array($_GET['tagssummary_timescale'], $this->Timescale)) {
+			$Configuration->object('tagssummary_timescale')->setValue($_GET['tagssummary_timescale']);
+			$Configuration->update('tagssummary_timescale');
+			Cache::delete(PluginConfiguration::CACHE_KEY);
+		}
+
+		$this->setConfiguration($Configuration);
+	}
+
+	/**
+	 * Method for getting the right symbol(s)
+	 */
+	protected function getRightSymbol() {
+		$CurrentSport = '';
+		$SportLinks = [];
+		$CurrentTimescale = '';
+		$TimescaleLinks = [];
+
+		foreach ($this->AllSports as $id => $name) {
+			$active = $id == (int)$this->Configuration()->value('sport');
+			$SportLinks[] = '<li'.($active ? ' class="active"' : '').'>'.Ajax::link($name, 'panel-'.$this->id(), Plugin::$DISPLAY_URL.'/'.$this->id().'?sport='.$id).'</li>';
+
+			if ($active) {
+				$CurrentSport = $name;
+			}
+		}
+
+		foreach (array_keys($this->Timescale) as $timescale) {
+			$active = $timescale == $this->Configuration()->value('tagssummary_timescale');
+			$name = $timescale.' '.($timescale != 1 ? __('months') : __('month'));
+			$TimescaleLinks[] = '<li'.($active ? ' class="active"' : '').'>'.Ajax::link($name, 'panel-'.$this->id(), Plugin::$DISPLAY_URL.'/'.$this->id().'?tagssummary_timescale='.$timescale).'</li>';
+
+			if ($active) {
+				$CurrentTimescale = $name;
+			}
+		}
+
+		$Links = '<li class="with-submenu"><span class="link">'.$CurrentSport.'</span>';
+		$Links .= '<ul class="submenu">'.implode('', $SportLinks).'</ul>';
+		$Links .= '</li>';
+
+		$Links .= '<li class="with-submenu"><span class="link">'.$CurrentTimescale.'</span>';
+		$Links .= '<ul class="submenu">'.implode('', $TimescaleLinks).'</ul>';
+		$Links .= '</li>';
+
+		return '<ul>'.$Links.'</ul>';
+	}
+
+	/**
+	 * Display the content
+	 * @see PluginPanel::displayContent()
+	 */
+	protected function displayContent() {
+		$Factory = \Runalyze\Context::Factory();
+		$Sport = $Factory->sport((int)$this->Configuration()->value('sport'));
+		$timescale = $this->Configuration()->value('tagssummary_timescale');
+
+		echo $this->getStyle();
+		echo '<div id="tags-summary">';
+		$this->showListFor($Sport, $timescale);
+		echo '</div>';
+
+		echo HTML::clearBreak();
+	}
+
+	/**
+	 * @param \Runalyze\Model\Sport\Entity $Sport
+	 */
+	protected function showListFor(Model\Sport\Entity $Sport, $timescale) {
+		$max = 0;
+		$countTagsForSport = DB::getInstance()
+			->query('SELECT activity_tag.tagid, tag.tag, COUNT(tagid) as `count`'.
+				' FROM '.PREFIX.'activity_tag activity_tag'.
+				' LEFT JOIN '.PREFIX.'training training ON training.sportid = '.$Sport->id().
```

Comment:
I just tried to use the plugin, but it fails after installation as there's only an empty sport object which causes an exception as `null == $Sport->id()`. (There wouldn't be any exception if `(int)$Sport->id()` is used, but that doesn't solve the main issue.)

Possible solutions: Check for this in `displayContent()` and use e.g. `Runalyze\Configuration::General()->mainSport()` (gives an id) or just the first entry of `$this->AllSports` as default - or show a message, that a sport type must be selected. I prefer to use a default one, but I don't care which one.